src/dropdown.ts

```diff
@@ -46,82 +50,255 @@ export class FdsDropdownEvent<T> extends CustomEvent<FdsDropdownOption<T>> {
+          class=${`ui-label-text ${this.getButtonCssClass()}`}
+          role="combobox"
+          aria-controls="options-list"
+          aria-expanded=${isFirstRender ? 'false' : this.getButton().ariaExpanded}
+        >
+          ${this.multiple ? multipleHeader() : singleHeader()}
+          <fds-icon icon="chevron-up"></fds-icon>
+          <fds-icon icon="chevron-down"></fds-icon>
+        </button>
+        ${optionsList}
+      </div>
     `
   }
 
+  setMultipleHeaderContent(): void {
+    const couterWidth = 30
+
+    const container = this.renderRoot.querySelector('.selected-options-container') as HTMLElement
+    const optionsEl = this.renderRoot.querySelector('.selected-options') as HTMLElement
+    const counterEl = this.renderRoot.querySelector('.overflow-counter') as HTMLElement
+
+    if (!container || !optionsEl || !counterEl) return
+
+    const optionTags = Array.from(optionsEl.querySelectorAll('.selected-tag'))
+    let hiddenCount = 0
+
+    const containerWidth = container.clientWidth - couterWidth
+    let currentWidth = 0
+
+    optionTags.forEach(tag => {
+      const tagEl = tag as HTMLElement
+      currentWidth += tagEl.offsetWidth
+      // offsetWidth doesn't take into account the fds-icon width. Check if the tag contains icon and add its width
+      const fdsIconEl = tagEl.querySelector('fds-icon')
+      if (fdsIconEl) {
+        currentWidth += parseInt(fdsIconEl.size.value)
+      }
+
+      if (currentWidth > containerWidth) {
+        tagEl.classList.add('hidden')
+        hiddenCount++
+      } else {
+        tagEl.classList.remove('hidden')
+      }
+    })
+
+    if (hiddenCount > 0) {
+      counterEl.classList.remove('hidden')
+      counterEl.textContent = `+${hiddenCount}`
+    } else {
+      counterEl.textContent = ''
+      counterEl.classList.add('hidden')
+    }
+  }
+
+  override updated(): void {
+    this.setMultipleHeaderContent()
+  }
+
   private handleKeypress(event: KeyboardEvent, selectedOption: FdsDropdownOption<T>): void {
     if (event.key === 'Enter') {
       this.handleSelect(selectedOption)
     }
   }
 
+  private getButton(): HTMLButtonElement {
+    const button = this.renderRoot.querySelector('button')
+
+    if (button === null) {
+      throw new Error('Button element not found')
+    }
+    return button
+  }
+
   private handleSelect(selectedOption: FdsDropdownOption<T>): void {
-    this._open = false
+    this.getButton().ariaExpanded = 'false'
     this.value = selectedOption
+    this.setValidity()
+    this.setFormValue()
     this.dispatchEvent(new FdsDropdownEvent(selectedOption))
   }
 
-  private getLabel(option?: FdsDropdownOption<T>): TemplateResult | null {
+  private handleMultiSelect(selectedOption: FdsDropdownOption<T>): void {
+    const allValues = this.getValues()
+    this.value = allValues.length > 0 ? allValues : undefined
+    this.setValidity()
+    this.setFormValue()
+    this.dispatchEvent(new FdsDropdownEvent(selectedOption))
```

Comment:
Nyt jokaisesta valinnasta lähetetään heti event yhdellä arvolla kun togglataan yksi arvoista. Mielestäni olisi parempi multiple vaihtoehdon tapauksessa lähettää koko dropdownin arvo (kaikki valinnat) kerralla vasta sitten kun dropdown suljetaan.

Mutta mun puolesta tämän parannuksen voisi tehdä myös myöhemmin erillisessä PR:ssä niin saadaan tämä eteenpäin nopeammin.

Reply:
Hyvä huomio. Mikäli mahdollista, niin voisin tehdä sen erillisessä PR:ssä